homeassistant/components/knx/climate.py

```diff
@@ -322,6 +357,40 @@ async def async_set_preset_mode(self, preset_mode: str) -> None:
             )
             self.async_write_ha_state()
 
+    @property
+    def fan_mode(self) -> str:
+        """Return the fan setting."""
+
+        fan_speed = self._device.current_fan_speed
+
+        if not fan_speed or self._attr_fan_modes is None:
+            return FAN_OFF
+
+        if self._device.fan_speed_mode == FanSpeedMode.STEP:
+            return self._attr_fan_modes[fan_speed]
+
+        # Find the closest fan mode percentage
+        closest_percentage = min(
+            self._fan_modes_percentages, key=lambda x: abs(x - fan_speed)
+        )
+        return self._attr_fan_modes[
+            self._fan_modes_percentages.index(closest_percentage)
+        ]
```

Comment:
```suggestion
        if not fan_speed or self._attr_fan_modes is None:
            return FAN_OFF

        if self._device.fan_speed_mode == FanSpeedMode.STEP:
            return self._attr_fan_modes[fan_speed]

        # Find the closest fan mode percentage
        closest_percentage = min(
            self._fan_modes_percentages[1:],  # fan_speed == 0 (FAN_OFF) is handled above
            key=lambda x: abs(x - fan_speed),
        )
        return self._attr_fan_modes[
            self._fan_modes_percentages.index(closest_percentage)
        ]
```

Reply:
Another way would be to not have the `0` item in `self._fan_modes_percentages` and handle `FAN_OFF` explicitly in `async_set_fan_mode(self, fan_mode: str)`.
However you prefer.

Reply:
removing zero makes sense, because the other use of _fan_modes_percentages, also omit the first element. Would have to add 1 when apply to the mode index.

Reply:
actually, missing around with the index is confusing, I would go what you suggested above.

Reply:
Fixed, I also added a test for the case.

Reply:
Looking at other integrations, it seems common to use a dict for that lookup and an inverted dict for the reverse-lookup instead of using lists with index. 
But for plain percent (no low-quiet-mode etc.) I guess index is fine.